test/ruby/test_autoload.rb

```diff
@@ -557,4 +558,26 @@ def test_autoload_parallel_race
       RUBY
     end
   end
+
+  def test_autoload_parent_namespace
+    Dir.mktmpdir('autoload') do |tmpdir|
+      autoload_path = File.join(tmpdir, "some_const.rb")
+      File.write(autoload_path, 'class SomeConst; end')
+
+      assert_separately(%W[-I #{tmpdir}], <<-RUBY)
+        $warnings = []
+        def Warning.warn(message)
+          $warnings << message
+        end
+
+        module SomeNamespace
+          autoload :SomeConst, #{File.realpath(autoload_path).inspect}
+          assert_not_nil SomeConst
+        end
+
+        assert_includes $warnings.first, "/some_const.rb to define SomeNamespace::SomeConst but it didn't"
+        assert_equal 1, $warnings.size
+      RUBY
```

Comment:
Why not `assert_warning`?

```suggestion
      assert_separately(%W[-I #{tmpdir} - #{File.realpath(autoload_path)}], <<-'RUBY')
        module SomeNamespace
          autoload :SomeConst, ARGV[0]
        end

        assert_warning(%r"/some_const\.rb to define SomeNamespace::SomeConst but it didn't\Z") do
          assert_not_nil(module SomeNamespace; SomeConst; end)
        end
      RUBY
```

Reply:
Because I didn't know it was a thing. Thanks!